### PR TITLE
[Delivers #107609788] exclude Delegates from attachment notifications

### PR DESCRIPTION
**Pull request by pkarman**
Story: https://www.pivotaltracker.com/story/show/107609788

Do not send attachment notification emails to proposal delegates, even if they relate to the currently actionable approval/step.

Testing guidelines are listed in the Story (link above).

### CODE REVIEW

File: app/models/proposal.rb [hunk 1]
```diff
@@ -99,6 +99,10 @@ def users
 
   alias_method :subscribers, :users
 
+  def users_except_delegates
```
Comment (jessieay) [inline app/models/proposal.rb]: unit test for this?

Comment (pkarman) [inline app/models/proposal.rb]: test added in 5ca40f4

File: spec/models/proposal_spec.rb [hunk 2]
```diff
@@ -112,6 +112,15 @@
...
       expect(proposal.users).to eq [observer]
     end
+
+    it "can exclude delegates" do
+      delegate_one = create(:user)
+      delegate_two = create(:user)
+      proposal = create(:proposal, :with_approver)
+      proposal.approvers.first.add_delegate(delegate_one)
+      proposal.approvers.first.add_delegate(delegate_two)
+      expect(proposal.users).to match_array(proposal.users_except_delegates + [delegate_one, delegate_two])
```
Comment (jessieay) [inline spec/models/proposal_spec.rb]: not trying to be a nit picker here, but if I were looking for a test for this method, I would expect it to be within a `describe "#method_name` block as opposed to embedded in a test for another method

Comment (pkarman) [inline spec/models/proposal_spec.rb]: nits picked in 4c73fbd

**Comment by jessieay** [comment]
code looks good! :ship:

**Comment by jessieay** [comment]
QA-ing now

**Comment by jessieay** [comment]
Ok, needed shared mailbox login to QA so am holding off for now but left a comment on the tracker story

**Comment by jessieay** [comment]
Passing this off to @phirefly for QA

**Comment by phirefly** [comment]
Validated on dev

**Pull request merged**